src/main/java/org/opentripplanner/model/TripTimeShort.java

```diff
@@ -13,50 +13,51 @@
 public class TripTimeShort {
 
     public static final int UNDEFINED = -1;
-    public FeedScopedId stopId;
-    public int stopIndex;
-    public int stopCount;
-    public int scheduledArrival = UNDEFINED ;
-    public int scheduledDeparture = UNDEFINED ;
-    public int realtimeArrival = UNDEFINED ;
-    public int realtimeDeparture = UNDEFINED ;
-    public int arrivalDelay = UNDEFINED ;
-    public int departureDelay = UNDEFINED ;
-    public boolean timepoint = false;
-    public boolean realtime = false;
-    public boolean isCancelledStop;
-    public RealTimeState realtimeState = RealTimeState.SCHEDULED ;
-    public long serviceDay;
-    public FeedScopedId tripId;
-    public String blockId;
-    public String headsign;
-    public int pickupType;
-    public int dropoffType;
+
+    private final FeedScopedId stopId;
+    private final int stopIndex;
+    private final int stopCount;
+    private final int scheduledArrival;
+    private final int scheduledDeparture;
+    private final int realtimeArrival;
+    private final int realtimeDeparture;
+    private final boolean isRecordedStop;
+    private final int arrivalDelay;
+    private final int departureDelay;
+    private final boolean timepoint;
+    private final boolean realtime;
+    private final boolean isCancelledStop;
+    private final RealTimeState realtimeState;
+    private final long serviceDay;
+    private final FeedScopedId tripId;
+    private final String blockId;
+    private final String headsign;
+    private final int pickupType;
+    private final int dropoffType;
 
     /**
      * This is stop-specific, so the index i is a stop index, not a hop index.
      */
-    public TripTimeShort(TripTimes tt, int i, Stop stop) {
+    public TripTimeShort(TripTimes tt, int i, Stop stop, ServiceDay sd) {
+        serviceDay         = sd != null ? sd.time(0) : UNDEFINED;
+        tripId             = tt.trip.getId();
         stopId             = stop.getId();
         stopIndex          = i;
         stopCount          = tt.getNumStops();
         scheduledArrival   = tt.getScheduledArrivalTime(i);
-        realtimeArrival    = tt.getArrivalTime(i);
-        arrivalDelay       = tt.getArrivalDelay(i);
         scheduledDeparture = tt.getScheduledDepartureTime(i);
-        realtimeDeparture  = tt.getDepartureTime(i);
-        departureDelay     = tt.getDepartureDelay(i);
         timepoint          = tt.isTimepoint(i);
         realtime           = !tt.isScheduled();
         isCancelledStop    = tt.isCancelledStop(i);
+        isRecordedStop     = tt.isRecordedStop(i);
```

Comment:
- The "is" prefix is normally not used in variable names, only in the "get" method. 
- Is "recorded" the proper term?  "realTimeIsActualTime"